test/Microsoft.AspNetCore.Blazor.Browser.Test/Interop/JavaScriptInvokeTests.cs

```diff
@@ -0,0 +1,429 @@
+
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.VoidParameterlessAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.VoidWithOneParameterAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.VoidWithTwoParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.VoidWithThreeParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.VoidWithFourParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.VoidWithFiveParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.VoidWithSixParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.VoidWithSevenParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.VoidWithEightParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.ReturnArrayAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.EchoOneParameterAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.EchoTwoParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.EchoThreeParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.EchoFourParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.EchoFiveParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.EchoSixParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.EchoSevenParametersAsync)));
+                result.Add(CreateMethodOptions(nameof(JavaScriptInterop.EchoEightParametersAsync)));
+
+                return result;
+
+                MethodOptions CreateMethodOptions(string methodName) =>
+                    new MethodOptions
+                    {
+                        Type = new TypeInstance
+                        {
+                            Assembly = typeof(JavaScriptInterop).Assembly.GetName().Name,
+                            TypeName = typeof(JavaScriptInterop).FullName
+                        },
+                        Method = new MethodInstance
+                        {
+                            Name = nameof(JavaScriptInterop.VoidParameterless)
+                        }
+                    };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(ResolveMethodPropertyData))]
+        public void ResolveMethod(object optionsObject)
+        {
+            var options = optionsObject as MethodOptions;
+
+            var resolvedMethod = options.GetMethodOrThrow();
+
+            Assert.NotNull(resolvedMethod);
+            Assert.Equal(options.Method.Name, resolvedMethod.Name);
+        }
+    }
+
+    internal class JavaScriptInterop
```

Comment:
Is this accidentally duplicated from `BasicTestApp`? I don't understand how it's being accessed from anywhere.

Reply:
Yeah, I haven't cleaned up.